packages/orbit-components/src/InputSelect/utils.ts

```diff
@@ -0,0 +1,50 @@
+import type { Option } from "./types";
+
+function separateGroupedAndUngrouped(options: Option[]): {
```

Comment:
Also, try this, looks more clean, I tried it, tests are passing: 

```
function separateGroupedAndUngrouped(options: Option[]) {
  const result = options.reduce<{ grouped: { [group: string]: Option[] }; ungrouped: Option[] }>(
    (acc, option) => {
      const { group } = option;

      if (group) {
        if (!acc.grouped[group]) {
          acc.grouped[group] = [];
        }
        acc.grouped[group].push(option);
      } else {
        acc.ungrouped.push(option);
      }

      return acc;
    },
    { grouped: {}, ungrouped: [] },
  );

  const grouped: Option[][] = Object.values(result.grouped);

  return { grouped, ungrouped: result.ungrouped };
```

Reply:
It does look more clean inside the reduce, but I honestly find it harder to read/understand the flow of the code. However, and since we can avoid multiple calls to `find`, I will go on with your suggestion, just changing the name of a variable for better readability